tests/test.py

```diff
@@ -49,84 +73,172 @@ def retry_unmount(path: str) -> None:
     else:
         raise RuntimeError(f"Failed to unmount {path}")
 
-
-class FatraceRunner:
-    def __init__(self, args: list[str]):
-        # we want to support multiple parallel FatraceRunners, so create our own private log dir
-        self.log_dir = tempfile.TemporaryDirectory()
-        self.output_file = os.path.join(self.log_dir.name, "fatrace.log")
-        self.log_content: str | None = None
-
+class FatraceRunner():
+    """Run text and json fatrace runners in parallel. args must include `--json`."""
+    def __init__(self, *args: str) -> None:
+        assert "--json" in args
+        # create our own private log dir
+        self.log_dir: tempfile.TemporaryDirectory[str] = tempfile.TemporaryDirectory()
+        self.text_output_file: str = os.path.join(self.log_dir.name, "fatrace.log.txt")
+        self.json_output_file: str = os.path.join(self.log_dir.name, "fatrace.log.json")
+        self.finished: bool = False
```

Comment:
OOI: These types are rather obvious, and mypy should be perfectly able to figure these out? Did you run into type errors, or is that more of a personal preference? (No big objection from me, just a bit unnecessary clutter)